File: py/encrypt.py
# ==============================================================================
#
#    Use:
#    encrypt("Hello World!", "key")
#    => b'WzM3NDQyLCA1MjQ4OCwgNTU4NzQsIDU2MDcyLCA1NzQxMCwgMTcxNjAsIDQ1MTIyLCA1NzYwOCwgNTg5NDYsIDU2MDcyLCA1MTc3OCwgMTc2NzJd'
#
# ==============================================================================

import base64

def encode(text):

        encoded_arr = [ord(i) for i in text]
        return encoded_arr

def encrypt(text, key):

        encoded_text = encode(text)
        encoded_key = encode(key)
        buf = []

        for x in encoded_text:
                for i in encoded_key:

                        x = x + 1 << i % 8

                encoded_key = list(reversed(encoded_key))
                buf.append(x)

        encrypt_string = base64.b64encode(str(buf).encode("ascii"))
        
        return encrypt_string

File: js/encrypt.js
/* ==========================================================================
 *
 *  Use:
 *  "Hello World!".encrypt("key")
 *  => "WzM3NDQyLDUyNDg4LDU1ODc0LDU2MDcyLDU3NDEwLDE3MTYwLDQ1MTIyLDU3NjA4LDU4OTQ2LDU2MDcyLDUxNzc4XQ=="
 *
 * ========================================================================== */

String.prototype.encode = function() {
    let array = [];
    for (let i of this) {
        array.push(i.charCodeAt(0))
    }
    return array
}
String.prototype.encrypt = function(key) {
    const encoded = this.encode();
    const keyEncoded = key.encode();
    // console.log(keyEncoded)
    let array = encoded.map(x => {
        x = parseInt(x)
        for (let i of keyEncoded) {
            x = x + 1 << i % 8
        }
		keyEncoded.reverse()
        return x;
    })
    return new Buffer(JSON.stringify(array)).toString('base64')
}

module.exports = (text, key) => text.encrypt(key)
